src/test/java/com/artipie/asto/StorageExtension.java

```diff
@@ -65,14 +69,23 @@
         .withSecureConnection(false)
         .build();
 
+    /**
+     * Vert.x instance to use in tests.
+     */
+    private Vertx vertx;
+
     @Override
     public void beforeAll(final ExtensionContext extension) {
         this.mock.beforeAll(extension);
+        this.vertx = Vertx.vertx();
     }
 
     @Override
     public void afterAll(final ExtensionContext extension) {
         this.mock.afterAll(extension);
+        if (this.vertx != null) {
+            this.vertx.rxClose().blockingAwait();
```

Comment:
@olegmoz It would be better if we manage `vertx` server disposal inside each method using `try-with-resource` construct, instead closing it manually. This also avoids the null test here.

Reply:
@paulodamaso it will make us duplicate Vert.x instance creation and disposing in every test, that is what I tried to avoid by keeping in `beforeAll`/`afterAll`. Also, it just impossible to pass something from test method to extension code as it runs actually before any test code, it generates tests.

Reply:
@olegmoz That's exactly my point: we must not share any data or objects between the tests. In these tests we are using the same vertx server for all tests, which is a bad practice. If you don't want to duplicate the code then don't use `beforeAll`/`afterAll`, use `beforeEach` / `afterEach`: this way we use a brand new vertx server instance for each test execution.

Reply:
@paulodamaso I see your point, but this will make test runs slower. Vert.x instance is basically a pool of threads and dispatcher, to me it does not seem as bigger problem then sharing JVM itself between tests and ForkJoinPool as part of it

Reply:
@g4s8 WDYT about it?

Reply:
@olegmoz how long it takes to start and stop vertx server?

Reply:
@g4s8 that's not server, it is `Vertx` instance. Start and stop is quite fast, no more then 10 ms